tests/integration/connector.py

```diff
@@ -37,3 +39,30 @@ def __exit__(self, exc_type, exc_val, exc_tb):
             self.connection.commit()
         self.cursor.close()
         self.connection.close()
+
+
+def create_db_connections(
+    num_connections: int, host: str, username: str, password: str, database: str
+) -> list[mysql.connector.MySQLConnection]:
+    """Create a list of database connections.
+
+    Args:
+        num_connections: Number of connections to create.
+        host: Hostname of the database.
+        username: Username to connect to the database.
+        password: Password to connect to the database.
+        database: Database to connect to.
+    """
+    connections = []
+    for _ in range(num_connections):
+        conn = mysql.connector.connect(
+            host=host,
+            user=username,
+            password=password,
+            database=database,
+            use_pure=True,
+        )
+        if conn.is_connected():
+            connections.append(conn)
+        sleep(0.5)
```

Comment:
sleep added to slow down tests so I could validate connections in database. Left here for the same reason, since the test is fairly quick

Reply:
It smells to me, it is a workaround for https://warthogs.atlassian.net/browse/DPE-5340 (we should sent peer details when we ready to accept traffic only).

Reply:
Not a workaround for that issue. Just to be able to observe the test while developing.
The error on the nightly tests seems a different issue to me, since the router is connecting to a standby cluster (and we have yet to change/discuss if the router should connect to database through router to pick leadership changes on async cases)